### PR TITLE
Closes #241: Filter Logs of KadaiConfiguration

### LINKED ISSUE #241: Filter Logs of KadaiConfiguration
In the log displaying all properties of Kadai Configuration, we only want to log kadai.* properties and also filter out any passwords or sensitive properties

### PR DESCRIPTION
**Pull request by jamesrdi**
### Thanks for your PR! Please fill out the following list :)

---

- [ ] I put the ticket or multiple tickets in review
- [x] Commit message format → Closes #&lt;Issue Number&gt; - Your commit message.
- [x] Sonarcloud link : [\<add the link here>](https://sonarcloud.io/summary/new_code?id=jamesrdi_kadai&branch=KAD-241)
- [x] No documentation update needed
- [ ] Link to PR with documentation update: \<add the link here>
- [ ] No Release Notes needed
- [ ] Release Notes :

```

```

### CODE REVIEW

File: lib/kadai-core/src/main/java/io/kadai/common/internal/KadaiEngineImpl.java [hunk 1]
```diff
@@ -128,14 +131,34 @@ public class KadaiEngineImpl implements KadaiEngine {
...
+  }
+
   protected KadaiEngineImpl(
       KadaiConfiguration kadaiConfiguration,
       ConnectionManagementMode connectionManagementMode,
       TransactionFactory transactionFactory)
       throws SQLException {
+    Builder builder = new Builder(kadaiConfiguration);
+    builder.setProperties(filterKadaiProperties(kadaiConfiguration.getProperties()));
+    KadaiConfiguration copiedKadaiConfiguration = builder.build();
```
Comment (CRoberto1926) [inline lib/kadai-core/src/main/java/io/kadai/common/internal/KadaiEngineImpl.java]: Why do you need to instantiate a whole new `KadaiConfiguration`?

Isn't it enough to just modify the `KadaiConfiguration.toString()`method to filter out the properties?

grafik

Comment (jamesrdi) [reply lib/kadai-core/src/main/java/io/kadai/common/internal/KadaiEngineImpl.java]: But the toString method can potentially be used by other methods, that require all properties unfiltered to be given as string

Comment (CRoberto1926) [reply lib/kadai-core/src/main/java/io/kadai/common/internal/KadaiEngineImpl.java]: But is it now used?

Comment (jamesrdi) [reply lib/kadai-core/src/main/java/io/kadai/common/internal/KadaiEngineImpl.java]: I don't think it is used right now yet

Comment (CRoberto1926) [reply lib/kadai-core/src/main/java/io/kadai/common/internal/KadaiEngineImpl.java]: I think that it is safe to assume that changing the output in the toString() method is not dangerous then, what do you think?

File: lib/kadai-core-test/src/test/java/acceptance/KadaiEngineInitalizationTest.java [hunk 2]
```diff
@@ -55,6 +67,46 @@ public OptionalInt calculatePriority(TaskSummary taskSummary) {
     }
   }
 
+  private static class TestKadaiEngineImpl extends KadaiEngineImpl {
```
Comment (CRoberto1926) [inline lib/kadai-core-test/src/test/java/acceptance/KadaiEngineInitalizationTest.java]: Why do you need this class and can't just call `KadaiEngineImpl.createKadaiEngine`?

File: lib/kadai-core-test/src/test/java/acceptance/KadaiEngineInitalizationTest.java [hunk 3]
```diff
@@ -20,23 +20,35 @@
...
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.TestInstance;
 import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.boot.test.system.CapturedOutput;
+import org.springframework.boot.test.system.OutputCaptureExtension;
 
+@ExtendWith(OutputCaptureExtension.class)
 @KadaiIntegrationTest
 public class KadaiEngineInitalizationTest {
```
Comment (CRoberto1926) [inline lib/kadai-core-test/src/test/java/acceptance/KadaiEngineInitalizationTest.java]: Can you fix the typo here please :)

`KadaiEngineInitializationTest`

**Review by CRoberto1926 — CHANGES_REQUESTED** [review]

File: lib/kadai-core-test/src/test/java/acceptance/KadaiEngineInitalizationTest.java (hunk 2 above)
Comment (jamesrdi) [reply lib/kadai-core-test/src/test/java/acceptance/KadaiEngineInitalizationTest.java]: You are right, I did not know we had that method

File: lib/kadai-core-test/src/test/java/acceptance/KadaiEngineInitalizationTest.java (hunk 3 above)
Comment (jamesrdi) [reply lib/kadai-core-test/src/test/java/acceptance/KadaiEngineInitalizationTest.java]: Yes I can

**Review by CRoberto1926 — APPROVED** [review]

**Pull request merged**